crates/bevy_render/src/render_resource/storage_buffer.rs

```diff
@@ -6,6 +6,21 @@ use encase::{
 };
 use wgpu::{util::BufferInitDescriptor, BindingResource, BufferBinding, BufferUsages};
 
+/// Stores data to be transferred to the GPU and made accessible to shaders as a storage buffer.
+///
+/// Storage buffers can be made available to shaders as some combination of read/write, unlike
+/// [`UniformBuffer`](crate::render_resource::UniformBuffer) which is read-only. Furthermore, storage buffers
+/// can store much larger data than uniform buffers, which are best suited to relatively small data.
+///
+/// Storage buffers can store runtime-sized arrays, but only if they are the last field in a structure. To store a
+/// runtime-sized array of data, use [`DynamicStorageBuffer`](crate::render_resource::DynamicStorageBuffer) instead.
```

Comment:
I think this needs to be improved. It’s complicated though so I’ll have to have a go at a suggestion another time.

Reply:
It's something like: if you want a runtime-sized array that you will index into from within your shaders, you can have one at the end of the inner type of a storage buffer. If you want multiple storage buffers of the same inner type and you will bind them in separate draw commands, you can use the dynamic storage buffer and pass the dynamic offsets when binding the buffer.

Dynamic storage buffers aren't really for runtime-sized arrays specifically. More for separating bindings due to some constraint or other or it just being more convenient for some reason. I kind of expect in practice it to be more desirable to use one storage buffer binding with a trailing runtime-sized array than to use dynamic offsets. This is because you can't batch draw calls together if you have to pass different dynamic offsets for the bindings.